app/services/service_listeners/attachment_link_header_updater.rb

```diff
@@ -0,0 +1,46 @@
+module ServiceListeners
+  class AttachmentLinkHeaderUpdater
+    include Rails.application.routes.url_helpers
+    include PublicDocumentRoutesHelper
+
+    attr_reader :attachment, :queue
+
+    def initialize(attachment, queue: nil)
+      @attachment = attachment
+      @queue = queue
+    end
+
+    def update!
+      return unless attachment.file?
+      attachment_data = attachment.attachment_data
+      return unless attachment_data.present?
+
+      # TODO: Fix the problem with this helper method appending .en to the URL
+      parent_document_url = public_document_url(attachment.attachable)
```

Comment:
Just in case you forgot about it, you might be able to use the `Whitehall::UrlMaker` here via `Whitehall.url_maker`.

Reply:
Thanks! I remembered running into this problem but couldn't remember what the fix was.